code/blake2/Hacl.Impl.Blake2.Generic.fst

```diff
@@ -1002,6 +1022,54 @@ let blake2_update #al #ms blake2_update_key blake2_update_blocks
       else blake2_update_blocks wv hash lb d)
     else blake2_update_blocks wv hash (size_to_limb al 0ul) d
 
+// TODO: this is technically redundant because output_len and key_len are ALREADY present in params. But this is good for regularity... or not?
```

Comment:
I'm a bit worried this will lead to mistakes, as this add extra preconditions to sync params with lengths.
I'd be in favor of expressing length constraints based on params.